spec/model/load_balancer_vm_port_spec.rb

```diff
@@ -123,22 +123,52 @@
           expect(session[:ssh_session]).to eq(second_ssh)
         end
 
-        it "is down if consecutive IOErrors are raised even on a stale connection" do
-          session[:last_pulse] = Time.now - 10
-
-          expect(session[:ssh_session]).to receive(:exec!).and_raise(IOError.new("closed stream"))
-
-          second_ssh = instance_double(Net::SSH::Connection::Session)
-          expect(second_ssh).to receive(:exec!).and_raise(IOError.new("closed stream"))
-          expect(lb_vm_port).to receive(:init_health_monitor_session).and_return(ssh_session: second_ssh)
-
-          expect(lb_vm_port.check_probe(session, :ipv6)).to eq("down")
+        shared_examples "a stale connection retrier" do |exception_class, exception_message|
```

Comment:
My personal preference is to avoid shared example and use a loop in the cases where it is simple to do so. In this case, you could iterate over a hash of exception class keys and exception message values.  However, that's up to you.

Reply:
I also was thinking of using a loop, but was wondering if rspec would do the right thing when running individual tests. Let me give it as hot.